### ISSUE TITLE
Some questions about checkpoints and learning rate

### ISSUE THREAD

**Issue by jovenwayfarer** [PyTorchLightning/pytorch-lightning#2864]
How to pass learning rate to progression bar and how to choose metric for saving model weights? Thanks!

**Comment by Borda** [comment]
@SkafteNicki mind have look? :]

**Comment by jovenwayfarer** [comment]
also how to print metrics and learning rate after each epoch? I tried to understand PrintCallback, but could not manage.

**Comment by rohitgr7** [comment]
to print metrics you can create a `Callback` and override `on_epoch_end` method. Inside that, you get the trainer parameter, use `trainer.lr_schedulers` to extract learning rates. To choose a metric for saving weights you have to assign the metric name to `monitor` parameter in `ModelCheckpoint`.

**Comment by jovenwayfarer** [comment]
@rohitgr7 Thanks! How to extract learning rate from `trainer.lr_schedulers`?

**Comment by rohitgr7** [comment]
I think it is either `trainer.lr_schedulers[0].optimizer.lr` or `trainer.lr_schedulers[0].optimizer.param_groups[0]['lr']`. Depends upon your optimizer param_group settings I think.
